fearless/Modules/Staking/ControllerAccount/ControllerAccountPresenter.swift

```diff
@@ -0,0 +1,201 @@
+        selectedAccount: AccountItem,
+        chain: Chain,
+        dataValidatingFactory: StakingDataValidatingFactoryProtocol,
+        logger: LoggerProtocol? = nil
+    ) {
+        self.wireframe = wireframe
+        self.interactor = interactor
+        self.viewModelFactory = viewModelFactory
+        self.applicationConfig = applicationConfig
+        chosenAccountItem = selectedAccount
+        self.chain = chain
+        self.dataValidatingFactory = dataValidatingFactory
+        self.logger = logger
+    }
+
+    private func updateView() {
+        guard
+            let stashItem = stashItem,
+            let accounts = accounts
+        else { return }
+        let viewModel = viewModelFactory.createViewModel(
+            stashItem: stashItem,
+            chosenAccountItem: chosenAccountItem,
+            accounts: accounts
+        )
+        canChooseOtherController = viewModel.canChooseOtherController
+        view?.reload(with: viewModel)
+    }
+
+    func refreshFeeIfNeeded() {
+        guard fee == nil, chosenAccountItem.address != stashItem?.stash else {
+            return
+        }
+
+        interactor.estimateFee(controllerAddress: chosenAccountItem.address)
+    }
+}
+
+extension ControllerAccountPresenter: ControllerAccountPresenterProtocol {
+    func setup() {
+        interactor.setup()
+    }
+
+    func handleControllerAction() {
+        guard canChooseOtherController else {
+            presentAccountOptions(for: stashItem?.controller)
+            return
+        }
+
+        guard let accounts = accounts else { return }
+        let context = PrimitiveContextWrapper(value: accounts)
+        let title = LocalizableResource<String> { locale in
+            R.string.localizable
+                .stakingControllerAccountTitle(preferredLanguages: locale.rLanguages)
+        }
+        wireframe.presentAccountSelection(
+            accounts,
+            selectedAccountItem: chosenAccountItem,
+            title: title,
+            delegate: self,
+            from: view,
+            context: context
+        )
+    }
+
+    func handleStashAction() {
+        presentAccountOptions(for: stashItem?.stash)
+    }
+
+    private func presentAccountOptions(for address: AccountAddress?) {
+        guard
+            let view = view,
+            let address = address
+        else { return }
+        wireframe.presentAccountOptions(
+            from: view,
+            address: address,
+            chain: chain,
+            locale: view.localizationManager?.selectedLocale ?? .current
+        )
+    }
+
+    func selectLearnMore() {
+        guard let view = view else { return }
+        wireframe.showWeb(
+            url: applicationConfig.learnControllerAccountURL,
+            from: view,
+            style: .automatic
+        )
+    }
+
+    func proceed() {
+        let locale = view?.localizationManager?.selectedLocale ?? Locale.current
+        DataValidationRunner(validators: [
```

Comment:
We need also fetch ```StoragePath.stakingLedger``` for the selected controller and show an error if it already exists.